### DISCUSSION TITLE
Codes for -SS, -DS, +FC

### DISCUSSION THREAD

**Discussion by henrychv** [wmo-im/iwxxm#322] [Q&A] [answered]
Hi,
I am a little bit lost.

There exists for `-SS, -DS, +FC` both definitions like http://codes.wmo.int/306/4678/-SS and http://codes.wmo.int/49-2/AerodromePresentOrForecastWeather/-SS

In iwxxm github in `IWXXM/rule/codes.wmo.int-49-2-AerodromePresentOrForecastWeather.rdf` is for all members, including `-SS, -DS, +FC` url `http://codes.wmo.int/306/4678/*` 

When I downloaded RDF from codes.wmo.int, in `AerodromePresentOrForecastWeather.rdf` is for all members except  `-SS, -DS, +FC`,  used same url `http://codes.wmo.int/306/4678/*`, but for  `-SS, -DS, +FC` is used newer url prefix `http://codes.wmo.int/49-2/AerodromePresentOrForecastWeather/*`

Do someone know a background behind this?

Should we expect, that there can come both codes in iwxxm document depending on decision/implementation time?

Is one of them invalid and it's use should lead to failing validation of the iwxxm document?

Thank you for any ideas/details.
Vaclav

**Comment by mgoberfield** [comment, marked as answer]
Hi Vaclav,

I am a little lost also as I am able to replicate what you are seeing.

The background is as follows:  The missing entries were discovered a while ago and [written](https://github.com/wmo-im/iwxxm-codelists/issues/2) up.  The 'turtle' [file](https://github.com/wmo-im/iwxxm-codelists/pull/5/commits/a7d842a9afb3babe88ea60fe17a724e924e62fc9) was updated to include them and reference the 306 entries as desired (The last entries on line 16 near the bottom).

And yet, when downloading the 49-2 container RDF file from the WMO Code Registry, these added entries don't reference 306 like the others do. (See and compare the "+SHSNGSRA" and "+SHGSRASN" entries surrounding the "+FC" entry)

![image](https://github.com/wmo-im/iwxxm/assets/7852621/b45873f8-42ac-44da-9c76-5882449c41a5)

My local copy AerodromePresentOrForecastWeather in RDF format (dated Feb 2022) does not have this problem.

I know the UKMet Office had to update the Code Registry software recently to fix an issue. I wonder if it went smoothly?

Perhaps @jkorosi or @blchoy can provide additional information.

UPDATE: I kindly suggest that you get the RDF files from the WMO IWXXM web site by downloading them from the [rule](https://schemas.wmo.int/iwxxm/2021-2/rule/) folder.

V/R,

mark

**Comment by blchoy** [comment]
Apologies for not replying earlier.  I am still away from office for a vacation.

I confirm the file [codes.wmo.int-49-2-AerodromePresentOrForecastWeather.rdf](https://github.com/wmo-im/iwxxm/files/11785670/AerodromePresentOrForecastWeather.zip) exported from the WMO Codes Registry contains the erroneous entries, viz:

```
    <skos:member>
      <skos:Concept rdf:about="http://codes.wmo.int/49-2/AerodromePresentOrForecastWeather/-DS">
        <dct:description xml:lang="en">Light duststorm</dct:description>
        <rdfs:label xml:lang="en">Light duststorm</rdfs:label>
        <reg:status>stable</reg:status>
        <skos:notation>-DS</skos:notation>
      </skos:Concept>
    </skos:member>
    ...
    <skos:member>
      <skos:Concept rdf:about="http://codes.wmo.int/49-2/AerodromePresentOrForecastWeather/+FC">
        <dct:description xml:lang="en">Well-developed funnel cloud(s) (tornado or water-spout)</dct:description>
        <rdfs:label xml:lang="en">Well-developed funnel cloud(s) (tornado or water-spout)</rdfs:label>
        <reg:status>stable</reg:status>
        <skos:notation>+FC</skos:notation>
      </skos:Concept>
    </skos:member>
    ...
    <skos:member>
      <skos:Concept rdf:about="http://codes.wmo.int/49-2/AerodromePresentOrForecastWeather/-SS">
        <dct:description xml:lang="en">Light sandstorm</dct:description>
        <rdfs:label xml:lang="en">Light sandstorm</rdfs:label>
        <reg:status>stable</reg:status>
        <skos:notation>-SS</skos:notation>
      </skos:Concept>
    </skos:member>
```

In <skos:Concept> the attribute rdf:about should contain ".../306/4678/..." instead of ".../49-2/AerodromePresentOfForecastWeather/...".

Checking also the respective RDF copies on https://schemas.wmo.int/iwxxm/2021-2/rule/ and https://schemas.wmo.int/iwxxm/2023-1/rule/ they have no problem at all.  Each time when we update code list entries we will cross check the RDF files we created with those downloaded from the WMO Codes Registry after deployment.  So this finding is likely occur some time after the publication of IWXXM 2021-2.  In any case as it does not involve a change of the content of the codes registry we could simply invite the registry administrator to make the necessary changes.

Could @marqh please kindly take a look?

Regards,
Choy
